interfaces.go

```diff
@@ -167,15 +204,30 @@ type Event struct {
 	Modules     map[string]string      `json:"modules,omitempty"`
 	Request     *Request               `json:"request,omitempty"`
 	Exception   []Exception            `json:"exception,omitempty"`
+	// Experimental: This is part of a beta feature of the SDK
+	StartTimestamp time.Time `json:"start_timestamp"`
+	// Experimental: This is part of a beta feature of the SDK
+	Spans []*Span `json:"spans,omitempty"`
 }
 
+// MarshalJSON converts the Event struct to JSON.
 func (e *Event) MarshalJSON() ([]byte, error) {
 	type alias Event
 	// encoding/json doesn't support the "omitempty" option for struct types.
 	// See https://golang.org/issues/11939.
 	// This implementation of MarshalJSON shadows the original Timestamp field
 	// forcing it to be omitted when the Timestamp is the zero value of
 	// time.Time.
+	if e.Timestamp.IsZero() && e.StartTimestamp.IsZero() {
```

Comment:
`StartTimestamp` should never be the zero value for valid transactions, and it should always be the zero value (and omitted) for errors.

Maybe we can make this more explicit and defensive here?

pseudo code:

```
if error event {
    always omit StartTimestamp
    always omit Spans
    always omit Type
    omit Timestamp if zero
}
if transaction {
   always include Timestamp and StartTimestamp (they should always be non-zero, server will refuse)
   maybe always include spans? Might be better to see `{spans:[]}` than the omission
   always include type: transaction
}
```

Reply:
Good catch, I completely missed this. I adjusted the `MarshalJSON` for the `Event` to check `e.Type == transactionType`, and then accordingly replicated logic similar to what you described above.